iosApp/iosApp/Modules/ImprovementProposals/ImprovementProposalsView.swift

```diff
@@ -29,48 +25,44 @@ final class ImprovementProposalsViewController: BaseViewController {
 
 extension ImprovementProposalsViewController: ImprovementProposalsView {
 
-    func update(with viewModel: ImprovementProposalsViewModel) {
+    func update(viewModel: ImprovementProposalsViewModel) {
         self.viewModel = viewModel
-        title = viewModel.title
-
-        switch viewModel {
-        case .loading:
-            showLoading()
-        case .loaded:
-            hideLoading()
-            collectionView.reloadData()
+        print("== ", viewModel)
+        if let loading = viewModel as? ImprovementProposalsViewModel.Loading {
+            refreshControl.beginRefreshing()
+        }
+        if let error = viewModel as? ImprovementProposalsViewModel.Error {
+            refreshControl.endRefreshing()
+            // TODO(PROPOSALS) Show alert
+        }
+        if let loaded = viewModel as? ImprovementProposalsViewModel.Loaded {
             refreshControl.endRefreshing()
```

Comment:
we need to find a cleaner way for this, prob better to be `if / else if / else if` for now